net_stats.h

```diff
@@ -0,0 +1,49 @@
+#ifndef _NET_STATS_H_
+#define _NET_STATS_H_
+#include <net/if.h>
+
+#include "postgres.h"
+
+typedef struct net_stat {
+	char name[IF_NAMESIZE + 1];
+	unsigned long long rx_bytes;
+	unsigned long long rx_packets;
+	unsigned long long rx_errors;
+	unsigned long long tx_bytes;
+	unsigned long long tx_packets;
+	unsigned long long tx_errors;
+	unsigned long long collisions;
+	unsigned long long saturation;
+
+	long long speed;
+	char duplex;
+
+	double rx_bytes_diff;
```

Comment:
I'm not sure why these are double, aren't they just numeric values, like a number of bytes/packets/etc?

Reply:
That's true, they could be numeric, but `S_VALUE ` macros, which is already used in a bunch of other places, explicitly cast the result to double.